src/drivers/dshot/DShot.cpp

```diff
@@ -299,21 +298,26 @@ int DShot::handle_new_bdshot_erpm(void)
 	int num_erpms = 0;
 	int telemetry_index = 0;
 	int erpm;
-	esc_status_s &esc_status = _telemetry->esc_status_pub.get();
+	esc_status_s &esc_status = esc_status_pub.get();
 
 	esc_status.timestamp = hrt_absolute_time();
 	esc_status.counter = _esc_status_counter++;
 	esc_status.esc_connectiontype = esc_status_s::ESC_CONNECTION_TYPE_DSHOT;
 	esc_status.esc_armed_flags = _outputs_on;
 
+	// We wait until all are ready.
+	if (up_bdshot_num_erpm_ready() < _num_motors) {
```

Comment:
On the IMXRT in an ideal case we receive from all motors the RPM.
If we miss the RPM from a single motor we still want to publish.
This if guard would stop that, but for logging you want to know this.

Reply:
Can you check the last commit? It should allow this check to always pass